test/python/circuit/test_controlled_gate.py

```diff
@@ -978,5 +978,57 @@ def test_ctrl_state_one(self, gate, controlled_gate):
         # Operator(controlled_gate.to_matrix()))
 
 
+@ddt
+class TestControlledGateLabel(QiskitTestCase):
+    """Tests for controlled gate labels."""
+    gates_and_args = [(XGate, []),
+                      (YGate, []),
+                      (ZGate, []),
+                      (HGate, []),
+                      (CXGate, []),
+                      (CCXGate, []),
+                      (C3XGate, []),
+                      (C4XGate, []),
+                      (MCXGate, [5]),
+                      (U1Gate, [0.1]),
+                      (CYGate, []),
+                      (CZGate, []),
+                      (CU1Gate, [0.1]),
+                      (SwapGate, []),
+                      (CCXGate, []),
+                      (RZGate, [0.1]),
+                      (RXGate, [0.1]),
+                      (RYGate, [0.1]),
+                      (CRYGate, [0.1]),
+                      (CRXGate, [0.1]),
+                      (CSwapGate, []),
+                      (U3Gate, [0.1, 0.2, 0.3]),
+                      (CHGate, []),
+                      (CRZGate, [0.1]),
+                      (CU3Gate, [0.1, 0.2, 0.3]),
+                      (MSGate, [5, 0.1]),
+                      (RCCXGate, []),
+                      (RC3XGate, []),
+                      (MCU1Gate, [0.1, 1]),
+                      (MCXGate, [5])
+                      ]
+
+    @data(*gates_and_args)
+    @unpack
+    def test_control_label(self, gate, args):
+        """Test gate(label=...).control(label=...)"""
+        cgate = gate(*args, label='a gate').control(label='a controlled gate')
+        self.assertEqual(cgate.label, 'a controlled gate')
+        self.assertEqual(cgate.base_gate.label, 'a gate')
+
+    @data(*gates_and_args)
+    @unpack
+    def test_control_label_1(self, gate, args):
+        """Test gate(label=...).control(1, label=...)"""
```

Comment:
Is there a case when num_ctrl_qubits != 1 which makes this test different than above?

Reply:
because `control` method has special cases for `num_ctrl_quits = 1`